### PR TITLE
Brand

### PR DESCRIPTION
**Pull request by jayesh-parmar**
Brand Task 
Brand Add , Brand Update and show

### CODE REVIEW

**Comment by jayesh-parmar** [comment]
👍

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    /**
+     * Update the specified resource in storage.
+     */
+    public function update(Request $request, string $id)
+    {
+        $request->validate([
+            'brand_name' => 'required',
+        ]);
+        $capitalizedString = ucfirst($request->brand_name);
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]: way this ?

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+namespace App\Http\Controllers;
+
+use App\Models\Brand;
+use Illuminate\Http\Request;
+
+class UserController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     */
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]: remove this comment

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    public function store(Request $request)
+    {
+        $request->validate([
+            'brand_name' => 'required',
+        ]);
+        $capitalizedString = ucfirst($request->brand_name);
+        $data= new Brand;
+        $data->brand_name = $capitalizedString;
+        $data->save();
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]: please check this to other way

File: resources/views/admin/pages/brands/brands.blade.php [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+<form action="{{route('store_brand')}}" method="post" class="mt-5">
+    @csrf
+    <div class="editor mx-auto w-10/12 flex flex-col text-gray-800 border border-gray-300 p-4 shadow-lg max-w-2xl">
+        <input 
+        class="title bg-gray-100 border border-gray-300 p-2 mb-4 outline-none" 
+        spellcheck="false" 
+        placeholder="Title" 
+        name="brand_name"
+         type="text">
```
Comment (dineshram7660) [inline resources/views/admin/pages/brands/brands.blade.php]:
```suggestion
        <input 
        	class="title bg-gray-100 border border-gray-300 p-2 mb-4 outline-none" 
        	spellcheck="false" 
        	placeholder="Title" 
        	name="brand_name"
         	type="text"
        />
```

File: resources/views/admin/pages/brands/brands.blade.php [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            <button>
+                <div class="btn border border-gray-300 p-2 px-4 font-semibold cursor-pointer text-gray-500 ml-auto">Add</div>
+            </button>
+        </div>
+    </div>
+</form>
+
+
+
+
```
Comment (dineshram7660) [inline resources/views/admin/pages/brands/brands.blade.php]: remove space

File: resources/views/admin/pages/brands/brands.blade.php [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+                </td>
+            </tr>
+            @endforeach
+
+        </tbody>
+    </table>
+</div>
+
+
+
```
Comment (dineshram7660) [inline resources/views/admin/pages/brands/brands.blade.php]: remove spance

File: resources/views/admin/pages/brands/brands.blade.php [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+</div>
+</div>
```
Comment (dineshram7660) [inline resources/views/admin/pages/brands/brands.blade.php]: div arrange properly

File: resources/views/admin/pages/brands/update.blade.php [hunk 8]
```diff
@@ -0,0 +1,31 @@
+@include('admin.layouts.main')
+
+
+<!-- component -->
```
Comment (dineshram7660) [inline resources/views/admin/pages/brands/update.blade.php]: remove

File: resources/views/dashboard.blade.php [hunk 9]
```diff
@@ -1,5 +1,6 @@
 @include('admin.layouts.main')
 
+
```
Comment (dineshram7660) [inline resources/views/dashboard.blade.php]:
```suggestion
```

File: routes/web.php [hunk 10]
```diff
@@ -14,9 +15,16 @@
...
+Route::get('/welcome', function () {
+    return view('welcome');
+});
+
 Route::get('/', function () {
     return view('auth.login');
 });
+// Route::get('/brands', function () {
+//     return view('admin.pages.brands');
+// })->name('brands');
```
Comment (dineshram7660) [inline routes/web.php]: remove

File: routes/web.php [hunk 11]
```diff
@@ -29,3 +37,13 @@
 });
 
 require __DIR__.'/auth.php';
+
+
+//brand route
+//--------------
```
Comment (dineshram7660) [inline routes/web.php]: remove

**Review by dineshram7660 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/UserController.php [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    public function store(Request $request)
+    {
+        $request->validate([
+            'brand_name' => 'required',
+        ]);
+        $data= new Brand;
+        $data->brand_name =Str::ucfirst($request->brand_name);
+        $data->save();
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]:
``` suggestion
       Brand::create([
			'brand_name' => $request->brand_name,
		]);
```

File: app/Http/Controllers/UserController.php [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        $data = Brand::all();
+        return view('admin.pages.brands.brands', ['data' => $data]);
+    }
+
+    public function store(Request $request)
+    {
+        $request->validate([
+            'brand_name' => 'required',
+        ]);
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]: create proper validation add =>  string, max:255, Unique

File: app/Http/Controllers/UserController.php [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+        $data= new Brand;
+        $data->brand_name =Str::ucfirst($request->brand_name);
+        $data->save();
+        return redirect()->route('brands')->with('success', $request->brand_name . 'New Brand successfully Added.');
+    }
+   
+    public function edit(string $id)
+    {
+        $data= Brand::find($id);
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]:
```suggestion
        $data = Brand::find($id);
```

File: app/Http/Controllers/UserController.php [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+        return view('admin.pages.brands.update',['data'=>$data]);
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]:
```suggestion
        return view('admin.pages.brands.update', ['data' => $data]);
```

File: app/Http/Controllers/UserController.php [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    public function update(Request $request, string $id)
+    {
+        $request->validate([
+            'brand_name' => 'required',
+        ]);
+    
+        $data= Brand::find($id);
+        $data->brand_name=Str::ucfirst($request->brand_name);
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]:
```suggestion
        $data->brand_name = $request->brand_name;
```

File: app/Http/Controllers/UserController.php [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+        return view('admin.pages.brands.update',['data'=>$data]);
+    }
+
+    public function update(Request $request, string $id)
+    {
+        $request->validate([
+            'brand_name' => 'required',
+        ]);
+    
+        $data= Brand::find($id);
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]:
```suggestion
        $brand = Brand::find($id);
```

File: app/Models/Brand.php [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use App\Traits\UuidTrait;
+
+
+class Brand extends Model
+{
+    use HasFactory, UuidTrait;
```
Comment (dineshram7660) [inline app/Models/Brand.php]: used HasUuids

File: app/Traits/UuidTrait.php [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace App\Traits;
+
+use Illuminate\Support\Str;
+
+trait UuidTrait
+{
+    /**
+     * Boot function from Laravel.
+     */
```
Comment (dineshram7660) [inline app/Traits/UuidTrait.php]: not required this

File: database/migrations/2023_08_28_063521_create_brands_table.php [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('brands', function (Blueprint $table) {
+            $table->uuid('id')->primary();
+            $table->string('brand_name');
```
Comment (dineshram7660) [inline database/migrations/2023_08_28_063521_create_brands_table.php:16]: add unique

File: database/migrations/2023_08_28_063521_create_brands_table.php [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     */
+    public function down(): void
+    {
+        Schema::dropIfExists('brands');
+    }
```
Comment (dineshram7660) [inline database/migrations/2023_08_28_063521_create_brands_table.php]: we can not used this so remove this

File: resources/views/admin/layouts/head.blade.php [hunk 22]
```diff
@@ -19,4 +20,8 @@
 </head>
 
 <body>
-    <div class="flex h-screen bg-gray-50 dark:bg-gray-900" :class="{ 'overflow-hidden': isSideMenuOpen }">
\ No newline at end of file
+    <div class="flex h-screen bg-gray-50 dark:bg-gray-900" :class="{ 'overflow-hidden': isSideMenuOpen }">
+
+
+
+
```
Comment (dineshram7660) [inline resources/views/admin/layouts/head.blade.php]: remove space

File: resources/views/admin/pages/brands/brands.blade.php [hunk 23]
```diff
@@ -0,0 +1,78 @@
+@include('admin.layouts.main')
+<br><br>
```
Comment (dineshram7660) [inline resources/views/admin/pages/brands/brands.blade.php]: why?

File: resources/views/admin/pages/brands/brands.blade.php [hunk 24]
```diff
@@ -0,0 +1,78 @@
...
+
+            <form action="{{route('store_brand')}}" method="post" class="mt-5">
+                @csrf
+                <div class="editor mx-auto w-10/12 flex flex-col text-gray-800 border border-gray-300 p-4 shadow-lg max-w-2xl">
+                    <input 
+                        class="title bg-gray-100 border border-gray-300 p-2 mb-4 outline-none" 
+                        spellcheck="false" 
+                        placeholder="Title" 
+                        name="brand_name"
+                        type="text">
```
Comment (dineshram7660) [inline resources/views/admin/pages/brands/brands.blade.php]:
```suggestion
                    <input 
                        class="title bg-gray-100 border border-gray-300 p-2 mb-4 outline-none" 
                        spellcheck="false" 
                        placeholder="Title" 
                        name="brand_name"
                        type="text"
                  />
```

Comment (dineshram7660) [reply resources/views/admin/pages/brands/brands.blade.php]: can create component for input?

File: routes/web.php [hunk 25]
```diff
@@ -29,3 +34,10 @@
 });
 
 require __DIR__.'/auth.php';
+
+Route::get('brands', [UserController::class,'index'])->name('brands');
+Route::post('brand',[UserController::class,'store'])->name('store_brand');
+Route::get('edit/{id}', [UserController::class, 'edit'])->name('edit');
+Route::post('update/{id}', [UserController::class, 'update'])->name('update_brand');
```
Comment (dineshram7660) [inline routes/web.php]: add in group

change URL and name also

Example
------------------

Route::controller(NotificationController::class)->name('notifications.')->group(function (): void {
    Route::get('fetch-notifications', 'fetchNotifications')->name('fetch');
    Route::post('mark-all-as-read', 'markAllAsRead')->name('mark_all_as_read');
});

**Review by dineshram7660 — CHANGES_REQUESTED** [review]

File: resources/views/admin/pages/brands/brands.blade.php (hunk 6 above)
Comment (dineshram7660) [reply resources/views/admin/pages/brands/brands.blade.php]: this is not done?

File: resources/views/admin/pages/brands/brands.blade.php (hunk 24 above)
Comment (dineshram7660) [reply resources/views/admin/pages/brands/brands.blade.php]: this not done

File: app/Http/Controllers/UserController.php [hunk 26]
```diff
@@ -0,0 +1,45 @@
...
+use App\Models\Brand;
+use Illuminate\Http\Request;
+use Illuminate\Support\Str;
+
+class UserController extends Controller
+{
+    public function index()
+    {
+        $data = Brand::all();
+        return view('admin.pages.brands.brands', ['data' => $data]);
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]:
```suggestion
        $brands = Brand::all();
        return view('admin.pages.brands.brands', ['brands' => $brands]);
```

File: app/Http/Controllers/UserController.php [hunk 27]
```diff
@@ -0,0 +1,45 @@
...
+
+    public function store(Request $request)
+    {
+        $request->validate([
+            'brand_name' => 'required|unique:brands,brand_name|max:255',
+        ]);
+        Brand::create([
+            'brand_name' => $request->brand_name, 
+        ]);
+        return redirect()->route('brands')->with('success', $request->brand_name . 'New Brand successfully Added.');
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]:
```suggestion
        return redirect()->route('brands')->with('success', 'Brand added successfully.');
```

File: app/Http/Controllers/UserController.php [hunk 28]
```diff
@@ -0,0 +1,45 @@
...
+    }
+   
+    public function edit(string $id)
+    {
+        $data = Brand::find($id);
+        return view('admin.pages.brands.update',['data' => $data]);
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]:
```suggestion
        $brand = Brand::find($brandId);
        return view('admin.pages.brands.update',['brand' => $brand]);
```

File: app/Http/Controllers/UserController.php [hunk 29]
```diff
@@ -0,0 +1,45 @@
...
+        $request->validate([
+            'brand_name' => 'required|unique:brands,brand_name|max:255',
+        ]);
+        Brand::create([
+            'brand_name' => $request->brand_name, 
+        ]);
+        return redirect()->route('brands')->with('success', $request->brand_name . 'New Brand successfully Added.');
+    }
+   
+    public function edit(string $id)
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]:
```suggestion
    public function edit(string $brandId)
```

File: app/Http/Controllers/UserController.php [hunk 30]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        $request->validate([
+            'brand_name' => 'required|unique:brands,brand_name|max:255',
+        ]);
+        $data = Brand::find($id);
+        $data->brand_name = $request->brand_name;
+        $data->save();
+
+        return redirect()->route('brands')->with('success', $request->brand_name.' Brand Name successfully Update.');
+    }
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]:
```suggestion
    public function update(Request $request, string $brandId)
    {
        $request->validate([
            'brand_name' => 'required|unique:brands,brand_name|max:255',
        ]);
        $brand = Brand::find($brandId);
        $brand->brand_name = $request->brand_name;
        $brand->save();

        return redirect()->route('brands')->with('success', 'Brand updated successfully.');
    }  
```

File: app/Models/Brand.php (hunk 18 above)
Comment (dineshram7660) [reply app/Models/Brand.php]: this is not done

**Review by dineshram7660 — CHANGES_REQUESTED** [review]

File: app/Traits/UuidTrait.php [hunk 31]
```diff
@@ -0,0 +1,41 @@
...
+
+    /**
+     * Get the auto-incrementing key type.
+     *
+     * @return string
+     */
+    public function getKeyType()
+    {
+        return 'string';
+    }
```
Comment (dineshram7660) [inline app/Traits/UuidTrait.php]:
```suggestion
    public function getKeyType(): string
    {
        return 'string';
    }
```

File: resources/views/admin/pages/brands/brands.blade.php [hunk 32]
```diff
@@ -0,0 +1,78 @@
...
+                />
+
+                @if($errors->has('brand_name'))
+                    <div class="font-bold text-white">{{ $errors->first('brand_name') }}</div>
+                @endif
+
+                <div class="buttons flex">
+                    <div class="btn border border-gray-300 p-1 px-4 font-semibold cursor-pointer text-gray-500 ml-10  ">Cancel</div>
+                    <button>
+                        <div class="btn border border-gray-300 p-2 px-4 font-semibold cursor-pointer text-gray-500 ml-auto">Add</div>
```
Comment (dineshram7660) [inline resources/views/admin/pages/brands/brands.blade.php]:
```suggestion
                        <div class="btn border border-gray-300 p-2 px-4 font-semibold cursor-pointer text-gray-500 ml-auto">
                        	Add
                        </div>
```

File: resources/views/admin/pages/brands/brands.blade.php [hunk 33]
```diff
@@ -0,0 +1,78 @@
...
+                        <th scope="col" class="px-6 py-3">
+                            Brand Name
+                        </th>
+                        <th scope="col" class="px-6 py-3">
+                            Action
+                        </th>
+                    </tr>
+                </thead>
+                <tbody>
+                    @foreach($data as $brand)
```
Comment (dineshram7660) [inline resources/views/admin/pages/brands/brands.blade.php]:
```suggestion
                    @foreach($brands as $brand)
```

**Review by dineshram7660 — CHANGES_REQUESTED** [review]

File: app/View/Components/Input.php [hunk 34]
```diff
@@ -0,0 +1,26 @@
...
+    public $name;
+    public $type;
+    public $placeholder;
+    public $value;
+    public function __construct($name,$type,$placeholder,$value)
+    {
+        $this->value = $value;
+       $this->name = $name;
+        $this->type = $type;
+        $this->placeholder = $placeholder;
```
Comment (dineshram7660) [inline app/View/Components/Input.php]:
```suggestion
        $this->value = $value;
       	$this->name = $name;
        $this->type = $type;
        $this->placeholder = $placeholder;
```

File: database/migrations/2023_08_28_063521_create_brands_table.php [hunk 35]
```diff
@@ -0,0 +1,16 @@
...
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+return new class extends Migration
+{
+    public function up(): void
+    {
+        Schema::create('brands', function (Blueprint $table) {
+            $table->uuid('id')->primary();
+            $table->string('brand_name');
```
Comment (dineshram7660) [inline database/migrations/2023_08_28_063521_create_brands_table.php]:
```suggestion
            $table->string('brand_name')->unique();
```

File: routes/web.php [hunk 36]
```diff
@@ -27,5 +27,13 @@
     Route::patch('/profile', [ProfileController::class, 'update'])->name('profile.update');
     Route::delete('/profile', [ProfileController::class, 'destroy'])->name('profile.destroy');
 });
+Route::controller(UserController::class)->middleware('auth')->name('brand.')->group(function () {
+    Route::get('brands', 'index')->name('brands');
+    Route::get('brands-add', 'addBrand')->name('add');
+    Route::post('brand-create', 'store')->name('store');
+    Route::get('edit-brand/{brandId}', 'edit')->name('edit');
+    Route::post('update-brand/{brandId}', 'update')->name('update');
```
Comment (dineshram7660) [inline routes/web.php]:
```suggestion
    Route::get('brands', 'index')->name('index');
    Route::get('brands/create', 'addBrand')->name('create');
    Route::post('brands', 'store')->name('store');
    Route::get('brands/{brandId}/edit', 'edit')->name('edit');
    Route::post('brands/{brandId}/update', 'update')->name('update');
```

**Review by dineshram7660 — CHANGES_REQUESTED** [review]

File: resources/views/admin/pages/brands/update.blade.php [hunk 37]
```diff
@@ -0,0 +1,24 @@
+@include('admin.layouts.main')
+<div class="heading text-center font-bold text-2xl m-5 text-gray-800 title bg-gray-100 border border-gray-300 p-2 mb-4 outline-none ">Update Brand </div>
+<form action="{{route('brand.update',$brands->id)}}" method="post">
+    @csrf
+    <div class="editor mx-auto w-10/12 flex flex-col text-gray-800 border border-gray-300 p-4 shadow-lg max-w-2xl">
+        <!-- <input class="title bg-gray-100 border border-gray-300 p-2 mb-4 outline-none" spellcheck="false" value="{{$brands->brand_name}}" name="brand_name" type="text" /> -->
```
Comment (dineshram7660) [inline resources/views/admin/pages/brands/update.blade.php]: remove not comment

File: resources/views/admin/pages/brands/update.blade.php [hunk 38]
```diff
@@ -0,0 +1,24 @@
...
+                <div class="btn border border-gray-300 p-1 px-4 font-semibold cursor-pointer text-gray-500 ml-auto">Update</div>
+            </button>
+        </div>
+    </div>
+</form>
+</div>
+</div>
+</body>
+
+</html>
```
Comment (dineshram7660) [inline resources/views/admin/pages/brands/update.blade.php]: is this a requiry or not?

File: resources/views/admin/pages/brands/brands.blade.php [hunk 39]
```diff
@@ -0,0 +1,52 @@
...
+                    </tr>
+                    @endforeach
+                </tbody>
+            </table>
+            {{ $brands->links() }}
+        </div>
+     </div>
+    </div>
+  </body>
+</html>
```
Comment (dineshram7660) [inline resources/views/admin/pages/brands/brands.blade.php]: is this a requiry or not?

File: resources/views/admin/pages/brands/add.blade.php [hunk 40]
```diff
@@ -0,0 +1,28 @@
...
+                    <button>
+                        <div class="btn border border-gray-300 p-2 px-4 font-semibold cursor-pointer text-gray-500 ml-auto">Add</div>
+                    </button>
+                </div>
+            </div>
+        </form>
+      </div>
+    </div>
+  </body>
+</html>
```
Comment (dineshram7660) [inline resources/views/admin/pages/brands/add.blade.php]: is this a requiry or not?

**Review by dineshram7660 — CHANGES_REQUESTED** [review]

File: resources/views/admin/pages/brands/add.blade.php (hunk 40 above)
Comment (jayesh-parmar) [reply resources/views/admin/pages/brands/add.blade.php]: dinesh bhai te me badhi vakhat chage kru chu but pachu hatu ne tevu thay jay che

File: resources/views/vendor/pagination/bootstrap-4.blade.php [hunk 41]
```diff
@@ -0,0 +1,56 @@
...
+    <span class="flex items-center col-span-3">
+        Showing 21-30 of 100
+    </span>
+    <span class="col-span-2"></span>
+    <!-- Pagination -->
+    <span class="flex col-span-4 mt-2 sm:mt-auto sm:justify-end">
+        @if ($paginator->hasPages())
+        <nav aria-label="Table navigation">
+            <ul class="inline-flex items-center">
+                {{-- Previous Page Link --}}
```
Comment (dineshram7660) [inline resources/views/vendor/pagination/bootstrap-4.blade.php]: please remove all comment code

File: resources/views/vendor/pagination/bootstrap-4.blade.php [hunk 42]
```diff
@@ -0,0 +1,56 @@
...
+                <li class="px-3 py-1 rounded-md focus:outline-none focus:shadow-outline-purple">
+                    <a class="page-link" href="{{ $paginator->previousPageUrl() }}" rel="prev" aria-label="@lang('pagination.previous')">&lsaquo;</a>
+                </li>
+                @endif
+
+                {{-- Pagination Elements --}}
+                @foreach ($elements as $element)
+                {{-- "Three Dots" Separator --}}
+                @if (is_string($element))
+                <li class="px-3 py-1 rounded-md focus:outline-none focus:shadow-outline-purple" aria-disabled="true"><span class="page-link">{{ $element }}</span></li>
```
Comment (dineshram7660) [inline resources/views/vendor/pagination/bootstrap-4.blade.php]: properly formate

**Review by dineshram7660 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/UserController.php [hunk 43]
```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Brand;
+use Illuminate\Http\Request;
+
+class UserController extends Controller
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]:
```suggestion

class BrandController extends Controller
```

File: database/migrations/2023_08_28_063521_create_brands_table.php [hunk 44]
```diff
@@ -0,0 +1,16 @@
...
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+return new class extends Migration
+{
+    public function up(): void
+    {
+        Schema::create('brands', function (Blueprint $table) {
+            $table->uuid('id')->primary();
+            $table->string('brand_name')->unique();
```
Comment (dineshram7660) [inline database/migrations/2023_08_28_063521_create_brands_table.php]:
```suggestion
            $table->string('name')->unique();
```

File: app/Http/Controllers/UserController.php [hunk 45]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    public function update(Request $request, string $brandId)
+    {
+        $brand = Brand::find($brandId);
+        if($brand->brand_name !== $request->brand_name ){
+            $request->validate([
+                'brand_name' => 'required|unique:brands,brand_name|max:255',
+            ]);
+        }
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]:
```suggestion
        $request->validate([
            'brand_name' => 'required|unique:brands,brand_name,'. $brandId .'|max:255',
        ]);

        $brand = Brand::find($brandId);
```

File: app/Http/Controllers/UserController.php [hunk 46]
```diff
@@ -0,0 +1,49 @@
...
+    }
+    public function addBrand()
+    {
+        return view('admin.pages.brands.add');
+    }
+
+    public function store(Request $request)
+    {
+        $request->validate([
+            'brand_name' => 'required|unique:brands,brand_name|max:255',
```
Comment (dineshram7660) [inline app/Http/Controllers/UserController.php]: change brand_name to name badhe change kari dejo

File: resources/views/admin/pages/brands/brands.blade.php [hunk 47]
```diff
@@ -0,0 +1,55 @@
...
+                Add New Brand
+            </button>
+        </a>
+    </div>
+    <div class="w-full overflow-x-auto  ">
+        <table class="w-full border-collapse ">
+            <thead>
+                <tr class="text-xs font-semibold tracking-wide text-left text-gray-500 uppercase border-b dark:border-gray-700 bg-gray-50 dark:text-gray-400 dark:bg-gray-800">
+                    <th class="px-4 py-3">Brand ID</th>
+                    <th class="px-4 py-3">Brand Name</th>
```
Comment (dineshram7660) [inline resources/views/admin/pages/brands/brands.blade.php]:
```suggestion
                    <th class="px-4 py-3">Id</th>
                    <th class="px-4 py-3">Name</th>
```

**Review by dineshram7660 — CHANGES_REQUESTED** [review]

File: resources/views/admin/pages/brands/update.blade.php [hunk 48]
```diff
@@ -0,0 +1,28 @@
...
+            @if($errors->has('brand_name'))
+              <span class="text-gray-700 dark:text-gray-400">{{ $errors->first('brand_name') }}</span>
+            @endif
+            <div class="mt-4">
+              <button class="px-4 py-2 bt-3 text-sm font-medium leading-5 text-white transition-colors duration-150 bg-purple-600 border border-transparent rounded-lg active:bg-purple-600 hover:bg-purple-700 focus:outline-none focus:shadow-outline-purple">
+                Update
+              </button>
+            </div>
+          </div>
+        </form>
```
Comment (dineshram7660) [inline resources/views/admin/pages/brands/update.blade.php]:
```suggestion
      <form action="{{route('brand.update',$brands->id)}}" method="post">
        @csrf
        <div class="px-4 py-3 mb-8 bg-white rounded-lg shadow-md dark:bg-gray-800">
          <label class="block text-sm">
            <span class="text-gray-700 dark:text-gray-400">Name</span>
            <x-input 
                placeholder="" 
                name="brand_name" 
                type="text" 
                value="{{$brands->brand_name}}" 
            />
          </label>
          @if($errors->has('brand_name'))
            <span class="text-gray-700 dark:text-gray-400">{{ $errors->first('brand_name') }}</span>
          @endif
          <div class="mt-4">
            <button class="px-4 py-2 bt-3 text-sm font-medium leading-5 text-white transition-colors duration-150 bg-purple-600 border border-transparent rounded-lg active:bg-purple-600 hover:bg-purple-700 focus:outline-none focus:shadow-outline-purple">
              Update
            </button>
          </div>
        </div>
      </form>
```

**Review by dineshram7660 — CHANGES_REQUESTED** [review]

**Pull request merged**